frontend/src/modules/textfilter/textfilter.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+import { FormField, Grid, Input, TextContent, TextFilterProps } from '@cloudscape-design/components';
+import { BaseKeyDetail, CancelableEventHandler } from '@cloudscape-design/components/internal/events';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import Condition from '../condition';
+import { debounce } from 'lodash';
+
+export type MLSTextFilterProps = TextFilterProps & {
+    onKeyDown?: CancelableEventHandler<BaseKeyDetail>,
+    requireEnter?: boolean,
+    delay?: number
+};
+
+const KEYCODE_ENTER = 13;
+const KEYCODE_ESC = 27;
+
+export function MLSTextFilter (props: MLSTextFilterProps) {
```

Comment:
I don't think this should exist. We need a very good reason for duplicating a built-in of Cloudscape. This opens a lot of doors for:
- Poorly maintained code
- More complex codebase
- Accessibility issues

We should not be in the business of overriding default table features unless we have a really good reason

Reply:
After further discussion, this allows us to better control submissions for the user and opens up more possibilities